File: src/very_easy/second_largest.zig
const std = @import("std");
const expectEqualSlices = std.testing.expectEqualSlices;
const expectEqual = std.testing.expectEqual;
const minInt = std.math.minInt;
pub fn secondLargest(comptime T: type, nums: []T) T {
    var first: T = minInt(T);
    var second: T = first;
    for (nums) |n| {
        if (first < n) {
            second = first;
            first = n;
        } else if (second < n and first > n) {
            second = n;
        }
    }
    return second;
}

File: src/main/php/VeryEasy/SecondLargest.php
<?php

declare(strict_types=1);

namespace Src\VeryEasy;

class SecondLargest
{
    public function solve(array $nums): int
    {
        $first = PHP_INT_MIN;
        $second = PHP_INT_MIN;
        foreach ($nums as $num) {
            if ($first === $num) {
                continue;
            }
            if ($first < $num) {
                $second = $first;
                $first = $num;
            } elseif ($second < $num && $first > $second) {
                $second = $num;
            }
        }

        return $second;
    }
}
